Do not lock down dev dependencies w/npm shrinkwrap; Bump various npm dependencies

jonasheinrich opened the pull request:
:warning: ~~**Includes commits from #2778, Please review, merge first.**~~

Continuation of #2661. Original PR #2400.

As far as I understood the commit mentioned in the title added dev dependencies to shrinkwrap. In a production setting, this can cause a lot of unnecessary traffic, e.g. when using the packaged based installation. So this PR aims to remove those dependencies. 
The packaged based installation currently works with a workaround: Just delete the shrinkwrap and compile the dependencies again. So if we delete the dependencies from shrinkwrap we do not need to delete the shrinkwrap anymore. That's also what this PR does.

Commit:
In packaging setup, do not remove shrinkwrap anymore

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://jonasheinrich-feature-revert-01efb3d.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://jonasheinrich-feature-revert-01efb3d.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

myabc committed:
Bump Bower to latest 1.4.1

See: https://github.com/bower/bower/releases/tag/v1.4.0

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Bump gulp patch version

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Peg gulp-livingstyleguide at 0.1.4

0.1.5 (re-)introduces a dependency on CoffeeScript:
https://github.com/efacilitation/gulp-livingstyleguide/commit/81f9178f84eeb596e72bb11ccdc91f0b7fe96514

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Rebuild npm-shrinkwrap.json

Attempt to deal with `npm WARN unmet dependency` errors.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add workaround for angular-busy template loading

Newer version of ngtemplate-loader are "cleverer" when it comes to
handling the relativeTo param:

    The path for file doesn't contains relativeTo param

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Mark broken inplaceEditor directive tests pending

Pending major refactoring.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Bump required Webpack version

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Commit:
Revert "Lock down dev dependencies w/npm shrinkwrap too"

This reverts commit 01efb3d58e9a1ffb3f72f2046ed6297debf48de7

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Bump karma-webpack to latest 1.5.0

This effectively reverts f5f51e6e67fa646b862098b97a8d15b7eea9a842

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Update shrinkwrap workflow in DEVELOP

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>
